src/entities/histogram/histogramImpl.js

```diff
@@ -0,0 +1,385 @@
+      this.currentMinuteMillis()
+    );
+  }
+
+  /**
+   * Get current time in millisecond.
+   * @returns {number}
+   */
+  currentClockMillis() {
+    return Date.now();
+  }
+
+  /**
+   * Get the timestamp of start of certain minute.
+   * @returns {number}
+   */
+  currentMinuteMillis() {
+    return Math.floor(this._clockMillis() / 60000) * 60000;
+  }
+
+  /**
+   * Add one value to the distribution.
+   * @param value
+   */
+  update(value) {
+    this.getCurrentBin().updateDist(value);
+  }
+
+  /**
+   * Bulk-update this histogram with a set of centroids.
+   * @param {Array} means - the centroid values
+   * @param {Array} counts - the centroid weights/sample counts
+   */
+  bulkUpdate(means, counts) {
+    this.getCurrentBin().bulkUpdateDist(means, counts);
+  }
+
+  /**
+   * Retrieve the current bin.
+   * @returns {MinuteBin} - Current minute bin
+   */
+  getCurrentBin() {
+    return this.getOrUpdateCurrentBin(this.currentMinuteMillis());
+  }
+
+  /**
+   * Get current minute bin.
+   * Will update _priorMinuteBinsList if the minute has passed.
+   * @param currMinuteMillis
+   * @returns {MinuteBin} MinuteBin of current minute.
+   */
+  getOrUpdateCurrentBin(currMinuteMillis) {
+    if (this._currentMinuteBin.minuteMillis === currMinuteMillis) {
+      return this._currentMinuteBin;
+    }
+    if (this._priorMinuteBinsList.length > this._MAX_BINS) {
+      this._priorMinuteBinsList.shift();
+    }
+    this._priorMinuteBinsList.push(this._currentMinuteBin);
+    this._currentMinuteBin = new MinuteBin(this._ACCURACY, currMinuteMillis);
+    return this._currentMinuteBin;
+  }
+
+  /**
+   * Return newly-updated _priorMinuteBinsList.
+   * @returns {[]|Array}
+   */
+  getPriorMinuteBinsList() {
+    this.getOrUpdateCurrentBin(this.currentMinuteMillis());
+    return this._priorMinuteBinsList;
+  }
+
+  /**
+   * Return the standard deviation of the values in the distribution.
+   */
+  stdDev() {
+    let mean = this.getMean();
+    let varianceSum = 0,
+      count = 0;
+    for (let minuteBin of this.getPriorMinuteBinsList()) {
+      let centroids = minuteBin.getCentroids();
+      count += centroids.reduce((accumulator, c) => accumulator + c.n, 0);
+      varianceSum += centroids.reduce(
+        (accumulator, c) => accumulator + c.n * (c.mean - mean) ** 2,
+        0
+      );
+    }
+    let variance = count === 0 ? 0 : varianceSum / count;
+    return Math.sqrt(variance);
+  }
+
+  /**
+   * Aggregate all the minute bins prior to the current minute.
+   * Returns a list of the distributions held within each bin.
+   * Note that invoking this method will also clear all data from the
+   * aggregated bins, thereby changing the state of the system and
+   * preventing data from being flushed more than once.
+   * @returns {Distribution[]}
+   */
+  flushDistributions() {
+    let distributions = [];
+    this.getOrUpdateCurrentBin(this.currentMinuteMillis());
+    this._priorMinuteBinsList.forEach(minuteBin =>
```

Comment:
it looks like you want this to be: const distributions = this._priorMinuteBinsList.map(x => x.toDistribution())